### PR TITLE
Task 7

### PR DESCRIPTION
**Pull request by Max-Korsakov**
See readme for the link to front end and tasks description
https://github.com/Max-Korsakov/nodejs-aws-be/blob/master/README.md

token name is Max_Korsakov, not Max-Korsakov because of syntax constraints of env variables (dash not allowed)

if you change the 'auth' token in the  local storage you will  receive a 403 error
if you delete the 'auth' key from local storage you will receive a 401 error

token set to local storage when you enter the product page

### CODE REVIEW

File: authorization-service/handlers/basicAuthorizer.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+) => {
+  if (event["type"] != "TOKEN") {
+    cb(new Error());
+  }
+  try {
+    const authToken = event.authorizationToken;
+    const encodedCreds = authToken.split(" ")[1];
+    const buff = Buffer.from(encodedCreds, "base64");
+    const plainCreds = buff.toString("utf-8").split(":");
+    const username = plainCreds[0];
```
Comment (mmarkelov) [inline authorization-service/handlers/basicAuthorizer.ts:48]: You can simplify this
```js
const [username, password] = plainCreds;
```

**Comment by mmarkelov** [comment]
@Max-Korsakov you should add some headers to your response. Cause right now there is an error 
Screen Shot 2020-11-30 at 11 07 03 AM

**Comment by Max-Korsakov** [comment]
> @Max-Korsakov you should add some headers to your response. Cause right now there is an error
>

Everything seems to work fine for me. Could you send me a full response (check it in the Firefox, Chome hides it).
image

**Comment by mmarkelov** [comment]
@Max-Korsakov 
```js
fetch("https://k68skd3820.execute-api.eu-west-1.amazonaws.com/dev/import?name=test.csv", {
  "headers": {
    "accept": "application/json, text/plain, */*",
    "accept-language": "en-US,en;q=0.9",
    "authorization": "Basic null",
    "sec-fetch-dest": "empty",
    "sec-fetch-mode": "cors",
    "sec-fetch-site": "cross-site"
  },
  "referrer": "https://d1j0a2cxkc2yn9.cloudfront.net/",
  "referrerPolicy": "strict-origin-when-cross-origin",
  "body": null,
  "method": "GET",
  "mode": "cors",
  "credentials": "include"
});
```

**Comment by Max-Korsakov** [comment]
No, I mean not headers, but info from the 'response' tab (вкладка ответ0

**Comment by mmarkelov** [comment]
@Max-Korsakov maybe you missed headers for **UnauthorizedException**
Screen Shot 2020-11-30 at 3 04 08 PM

**Comment by Max-Korsakov** [comment]
seems currently everything works correctly.

**Comment by mmarkelov** [comment]
@Max-Korsakov yeah! That's ok now. But I got **401** error instead of **403** for wrong token

File: authorization-service/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "name": "import-service",
+  "version": "1.0.0",
+  "description": "Serverless webpack example using Typescript",
+  "main": "handler.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "dependencies": {
+    "aws-lambda": "^1.0.6",
+    "aws-sdk-mock": "^5.1.0",
```
Comment (Ilya-Valasiuk) [inline authorization-service/package.json:11]: do we need it in `dependencies `?

Also lambda under the hood has `aws-sdk`, so you should move `aws-lambda` to devDependencies

**Pull request merged**